### ISSUE TITLE
script sort with "collapse" failed to get_ score variable

### ISSUE THREAD

**Issue by yanchaoguo** [elastic/elasticsearch#87772]
### Elasticsearch Version

8.2.0

### Java Version

openjdk version "18" 2022-03-22

### OS Version

Linux product-es-v2-001.eqxiu.com 3.10.0-1160.62.1.el7.x86_64 #1 SMP Tue Apr 5 16:57:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Problem Description

![错误截图](https://media.githubusercontent.com/avatars/24366668?orig=1&token=AC2HUWS7BS3I2IFEDXIL6KLCVP5OM)
I can't get _score value when using script sort for custom sorting , I have set track_ scores=true ; When I remove the collapse part of the query, I can get a "_score" . I can add a sorting dimension like the following

GET search_ data/_ search

{

"_source": ["product_quality_score"],

"collapse": {

"field": "gid.keyword"

},

"track_scores":true,

"sort": [

{

"_script": {

"order": "desc",

"type": "number",

"script": {

"lang": "painless",

"source": """

if(doc['product_quality_score'].size()>0)

_ score+doc['product_quality_score']. value;

else

_ score

"""

}

}

},

"price"

],

"query": {

"bool": {

"must": [

{

"match": {

"Title2": "wedding invitation"

}

}

]

}

}

}

### Steps to Reproduce

none

**Comment by benwtrent** [comment]
This works in the latest version. I have tested the included query:

```
GET search_data/_search
{
  "_source": [
    "product_quality_score"
  ],
  "collapse": {
    "field": "gid.keyword"
  },
  "track_scores": true,
  "sort": [
    {
      "_script": {
        "order": "desc",
        "type": "number",
        "script": {
          "lang": "painless",
          "source": """
if(doc['product_quality_score'].size()>0)
_score+doc['product_quality_score'].value;
else
_score
"""
        }
      }
    }
  ],
  "query": {
    "bool": {
      "must": [
        {
          "match": {
            "Title2": "wedding invitation"
          }
        }
      ]
    }
  }
}
```

With some sample data and verified that the sorting is based on either the `_score` directly or some value multiplied by the `_score`

**Comment by rschoenbeck** [comment]
I'm having the same issue on Elasticsearch 8.7.1 (Spring Data Elasticsearch 5.1). 

In case anyone else is searching about this issue and wondering, it seems like this is fixed in the ES version available on the next Spring Data ES release (ES 8.11.1 / Spring Data ES 5.2.x)
